### PR TITLE
unistore: get/batchGet/scan support read-through-lock

### LINKED ISSUE #29881: unistore supports read-through-lock
## Enhancement

unistore has to implement it as TiKV does https://github.com/tikv/tikv/pull/11238 to write tests.

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve?

Issue Number: close #29881

Problem Summary:

unistore needs to support read-through-lock to write test.

### What is changed and how it works?

get/batchGet/scan supports it. However, coprocessor in unistore is much different from TiKV -- unistore scans all locks in the range at first, while TiKV scans key by key and check lock. I can't find a suitable way to implement read-through-lock for coprocessor in unistore, so I just implement it get/batchGet/scan. 

### Check List

Tests 

- [-] Unit test

Side effects

Documentation

### Release note

```release-note
None
```

### CODE REVIEW

File: store/mockstore/unistore/tikv/mvcc.go [hunk 1]
```diff
@@ -1386,14 +1408,32 @@ func (store *MVCCStore) DeleteFileInRange(start, end []byte) {
...
+func (store *MVCCStore) Get(reqCtx *requestCtx, key []byte, version uint64) ([]byte, error) {
+	lockPairs, err := store.CheckKeysLock(version, reqCtx.rpcCtx.ResolvedLocks, reqCtx.rpcCtx.CommittedLocks, key)
+	if err != nil {
+		return nil, err
+	}
+	if len(lockPairs) != 0 {
+		return getValueFromLock(lockPairs[0].lock), nil
+	}
+	reader := reqCtx.getDBReader()
+	return reader.Get(key, version)
```
Comment (sticnarf) [inline store/mockstore/unistore/tikv/mvcc.go:1421]: Do we need to deep copy it?

Comment (youjiali1995) [reply store/mockstore/unistore/tikv/mvcc.go]: oh, yes.

**Review by sticnarf — APPROVED** [review]
LGTM. Please update the kvproto.

**Comment by sticnarf** [comment]
A assertion type mismatch needs be fixed...

File: store/mockstore/unistore/tikv/mvcc.go [hunk 2]
```diff
@@ -1386,14 +1408,32 @@ func (store *MVCCStore) DeleteFileInRange(start, end []byte) {
...
 	pairs := make([]*kvrpcpb.KvPair, 0, len(keys))
 	remain := make([][]byte, 0, len(keys))
 	for _, key := range keys {
-		err := store.CheckKeysLock(version, reqCtx.rpcCtx.ResolvedLocks, key)
+		lockPairs, err := store.CheckKeysLock(version, reqCtx.rpcCtx.ResolvedLocks, reqCtx.rpcCtx.CommittedLocks, key)
 		if err != nil {
 			pairs = append(pairs, &kvrpcpb.KvPair{Key: key, Error: convertToKeyError(err)})
+		} else if len(lockPairs) != 0 {
+			value := getValueFromLock(lockPairs[0].lock)
+			if value != nil {
```
Comment (you06) [inline store/mockstore/unistore/tikv/mvcc.go:1434]: If the lock type is not `Put`, value will be nil, so we skip this key here.

Comment (youjiali1995) [reply store/mockstore/unistore/tikv/mvcc.go:1434]: The lock type is either `Put` or `Del` because only locks that can block read are in `LockPair`.

Comment (you06) [reply store/mockstore/unistore/tikv/mvcc.go:1434]: I mistakenly thought the `Del` record needs a `nil` return, LGTM.

**Review by you06 — APPROVED** [review]
LGTM

**Pull request merged**